types/vast-client/index.d.ts

```diff
@@ -82,136 +97,198 @@ export class tracker {
+    removeURLTemplateFilter(): void;
     /**
-     * Remove the specified listener for the event named eventName.
+     * Reset URLTemplateFilters to empty, previous replace function set with addURLTemplateFilter() are no longer called.
      */
-    off(eventName: string, listener: (error: VastError) => void): void;
+    clearUrlTemplateFilters(): void;
     /**
-     * Remove replace function from URLTemplateFilters array.
-     * Replace function won't be called on the next VAST URL encountred by the parser.
+     * Returns how many replace function are set (ie: URLTemplateFilters length)
      */
-    removeURLTemplateFilter(cb: (vastUrl: string) => string): void;
+    countURLTemplateFilters(): number;
+    /**
+     * Tracks the error provided in the errorCode parameter and emits a VAST-error event for the given error.
+     */
+    trackVastError(
+        /**
+         * An Array of url templates to use to make the tracking call
+         */
+        urlTemplates: string[],
+        errorCode: Pick<VastError, 'ERRORCODE'>,
+        ...data: Array<Pick<VastError, Exclude<keyof VastError, 'ERRORCODE'>>>,
+    ): void;
+    /**
+     * Fetches a VAST document for the given url.
+     * Returns a Promise which resolves with the fetched xml or rejects with an error, according to the result of the request.
+     */
+    fetchVAST(
+        /**
+         * The url to request the VAST document.
+         */
+        url: string,
+        /**
+         * how many times the current url has been wrapped
+         */
+        wrapperDepth?: number,
+        /**
+         * url of original wrapper
+         */
+        originalUrl?: string
+    ): Promise<Document>;
+    /**
+     * Fetches and parses a VAST for the given url.
+     * Returns a Promise which resolves with a fully parsed VASTResponse or rejects with an Error.
+     */
+    getAndParseVAST(
+        /**
+         * The url to request the VAST document.
+         */
+        url: string,
+        /**
+         * An optional Object of parameters to be used in the parsing process.
+         */
+        options?: VastRequestOptions,
+    ): Promise<VastResponse>;
+    /**
+     * Parses the given xml Object into a VASTResponse.
+     * Returns a Promise which either resolves with the fully parsed VASTResponse or rejects with an Error.
+     */
+    parseVAST(
+        /**
+         * A VAST XML document
+         */
+        vastXml: Document,
+        /**
+         * An optional Object of parameters to be used in the parsing process.
+         */
+        options?: VastRequestOptions,
+    ): Promise<VastResponse>;
+}
+
+export interface VASTClientCustomStorage {
+    getItem(key: string): string | null;
+    setItem(key: string, val: string): void;
+    [key: string]: any | (() => any);
+}
+
+export function UrlHandlerCbType(err: null, xml: XMLDocument): void;
+export function UrlHandlerCbType(err: Error): void;
```

Comment:
why not `export function UrlHandlerCbType(err: Error | null, xml: XMLDocument): void;`

Reply:
because this way you dont have to specify xml if there is an error